internal/controller/promotions/promotions.go

```diff
@@ -401,12 +401,11 @@ func (r *reconciler) promote(
 		return nil
 	}
 
-	var targetFreightIndex int
 	var targetFreight *kargoapi.Freight
-	for i, availableFreight := range stage.Status.AvailableFreight {
+	for _, availableFreight := range stage.Status.AvailableFreight {
 		if availableFreight.ID == freightID {
-			targetFreightIndex = i
 			targetFreight = availableFreight.DeepCopy()
+			targetFreight.Qualified = false
```

Comment:
This disqualifies current Freight. The next Stage reconciliation will qualify it if the Stage is healthy.